elyra/pipeline/pipeline_definition.py

```diff
@@ -346,14 +363,46 @@ def remove_env_vars_with_matching_secrets(self):
         In the case of a matching key between env vars and kubernetes secrets,
         prefer the Kubernetes Secret and remove the matching env var
         """
-        if self.get_component_parameter(ENV_VARIABLES) and self.get_component_parameter(KUBERNETES_SECRETS):
+        env_vars = self.get_component_parameter(ENV_VARIABLES)
+        secrets = self.get_component_parameter(KUBERNETES_SECRETS)
+        if env_vars and isinstance(env_vars, KeyValueList) and secrets and isinstance(secrets, KeyValueList):
             new_list = KeyValueList.difference(
                 minuend=self.get_component_parameter(ENV_VARIABLES),
                 subtrahend=self.get_component_parameter(KUBERNETES_SECRETS),
             )
             if new_list:
                 self.set_component_parameter(ENV_VARIABLES, new_list)
```

Comment:
Sorry. I just noticed the help-string on the method and didn't realize it was that close to this block, so the comment portion of this suggestion isn't necessary.

Reply:
Agree on the necessity of a comment!

> Hmm, I don't think we want the check against a Falsey result. Since None can't be returned, we should unconditionally reset ENV_VARIABLES to the empty list in the case when there are only secrets present.

I must have had this check in here because [set_component_parameter](https://github.com/elyra-ai/elyra/blob/c31164f5fa1ca354cc1e3ef43e48fd790e790302/elyra/pipeline/pipeline_definition.py#L315-L316) actually throws a `ValueError` if you send it a Falsey value of any type. I'm not sure if there's a reason for this that I can't think of but it seems pretty unnecessary

Reply:
Thoughts on changing that `set_component_parameter` line to `if value is None`?

Reply:
I see. Of course, I didn't check that prior to my comment.  😅 

I think the check in `set_component_parameter` should be updated to `value is not None` and "empty values" should be allowed assuming they don't cause issues downstream.   It looks like Falseys are tolerated in these structures based on existing calls to `get_component_parameter()` so I suspect this should be okay.

I wonder if _any_ value "kind" should be tolerated (`None`, `""`, `[]`, etc.) so long as the `key` has a legitimate (non-falsey) value.  Then, it's really up to the consumer (where it matters) to make a decision regarding `None` and other falsey conditions.  This may be outside the scope of this PR, but I think, from a general design model, that makes sense (and would eliminate lots of these kinds of checks).